### PR TITLE
feat(Wolfy): add multiLanguage support & handle event feature

### PR DESCRIPTION
**Pull request by Tenebrosful**
**Describe the changes in this pull request:**

- All strings are now dynamic and change with Wolfy's lang
- Handling Event new Wolfy feature

<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<br>

image

image

image

</details>

### CODE REVIEW

File: websites/W/Wolfy/presence.ts [hunk 1]
```diff
@@ -1,3 +1,106 @@
...
+		visitHelpCenter: "Visite le centre d'aide",
+		visitProfile: "Voir le profil de $username",
+		visitWolfy: "Visiter le site de Wolfy",
+		waiting: "EN ATTENTE"
+	}
+};
+
+function getString(key: keyof langStrings) {
+	return (
+		strings[document.querySelector("html")?.lang || "en"]?.[key] ||
```
Comment (ImRodry) [inline websites/W/Wolfy/presence.ts:98]:
```suggestion
		strings[document.querySelector("html")?.lang ?? "en"]?.[key] ||
```

File: websites/W/Wolfy/presence.ts [hunk 2]
```diff
@@ -54,63 +160,75 @@ async function addJoinGameButton(presenceData: PresenceData, gameId: string) {
...
-	let label = `Visiter le profil de ${username}`;
+	let label = `${getString("visitProfile").replace("$username", username)}`;
 	if (label.length > 32) label = `${label.slice(0, 31)}…`;
 
 	addButton(presenceData, label, `https://wolfy.fr/leaderboard/${username}`);
 }
 
 function addVisitWolfyButton(presenceData: PresenceData) {
-	addButton(presenceData, "Visiter le site de Wolfy", "https://wolfy.fr");
+	addButton(presenceData, getString("visitWolfy"), "https://wolfy.net");
```
Comment (ImRodry) [inline websites/W/Wolfy/presence.ts:194]: Should probably check for the user's language and redirect them to the appropriate website or use document.location.origin

File: websites/W/Wolfy/presence.ts [hunk 3]
```diff
@@ -178,66 +303,87 @@ presence.on("UpdateData", async () => {
 			currTime = document.querySelector("div.Header_timer__36MsP").textContent;
 		}
 
+		const [startTimestamp, endTimestamp] = getInGameTimestamps(cp, currTime);
```
Comment (ImRodry) [inline websites/W/Wolfy/presence.ts:306]:
```suggestion
```

File: websites/W/Wolfy/presence.ts [hunk 4]
```diff
@@ -178,66 +303,87 @@ presence.on("UpdateData", async () => {
...
+
+		presenceData.details = getString("playing");
+
+		presenceData.smallImageKey = "live";
+		presenceData.smallImageText = getString("playing");
+		if (currTime && currTime.includes(":")) {
+			presenceData.startTimestamp = startTimestamp;
+			presenceData.endTimestamp = endTimestamp;
```
Comment (ImRodry) [inline websites/W/Wolfy/presence.ts:314]:
```suggestion
		if (currTime?.includes(":")) {
			[presence.startTimestamp, presence.endTimestamp] = getInGameTimestamps(cp, currTime);
```

File: websites/W/Wolfy/presence.ts [hunk 5]
```diff
@@ -178,66 +303,87 @@ presence.on("UpdateData", async () => {
...
-				presenceData.state = "Consulte les dernières actualités";
+			case "articles":
+				presenceData.state = getString("consultingArticles");
 				presenceData.smallImageKey = "reading";
-				presenceData.smallImageText = "En train de lire";
+				presenceData.smallImageText = getString("consultingArticles");
 				break;
-			case "/play":
-			case "/":
+			case "play":
```
Comment (ImRodry) [inline websites/W/Wolfy/presence.ts:384]: no need to have a case if it uses the same as default
```suggestion
```

**Review by ImRodry — CHANGES_REQUESTED** [review]

File: websites/W/Wolfy/presence.ts (hunk 2 above)
Comment (Tenebrosful) [reply websites/W/Wolfy/presence.ts:194]: Wolfy is not officially Wolfy.net, wolfy.fr always redirect to Wolfy.net

**Comment by Tenebrosful** [comment]
(I will commit prettier lint tomorrow)

File: websites/W/Wolfy/presence.ts (hunk 2 above)
Comment (Tenebrosful) [reply websites/W/Wolfy/presence.ts:194]: Also it use to always link to wolfy.net even thought we are browsing help.wolfy.net

File: websites/W/Wolfy/presence.ts [hunk 6]
```diff
@@ -1,3 +1,106 @@
...
+		readingAnArticle: "Lit l'article ⤵️",
+		toConsultArticle: "Consulter l'article",
+		toConsultCategory: "Consulter la catégorie",
+		toConsultEvent: "Consulter l'événement",
+		visitHelpCenter: "Visite le centre d'aide",
+		visitProfile: "Voir le profil de $username",
+		visitWolfy: "Visiter le site de Wolfy",
+		waiting: "EN ATTENTE"
+	}
+};
```
Comment (EncryptedDev) [inline websites/W/Wolfy/presence.ts:94]: We do not allow multiLang to be added in this way
> Presences should use [common details](https://api.premid.app/v2/langFile/presence/en) (strings starting with "general."). You can achieve this using multiLanguage with the provided strings. If your presence requires custom strings, then you shouldn't use multiLanguage until the presence gets 1000 users. You can find an example [here](https://docs.premid.app/dev/presence/class#getstringsobject).

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/W/Wolfy/presence.ts (hunk 6 above)
Comment (Tenebrosful) [reply websites/W/Wolfy/presence.ts:94]: well that's pretty sad, it was suppose to come the release of Wolfy english games.

I guess i will just make the fix and event handle then :)

**Pull request closed without merging**